std/engine/face/stream_face.go

```diff
@@ -35,10 +35,6 @@ func (f *StreamFace) String() string {
 	return fmt.Sprintf("stream-face (%s://%s)", f.network, f.addr)
 }
 
-func (f *StreamFace) Trait() Face {
```

Comment:
To clarify: this `Trait` is to prevent uncertainty of interface type checking in Go: when the implementation broke an interface, it is the caller rather than the implementation that raises errors, and the error message sometimes confusing.

Reply:
Yeah, it's really annoying Go doesn't have `implements`. I can't remember why I removed this, will likely restore it.